config.go

```diff
@@ -69,6 +73,28 @@ func (c *Config) AddMembershipListener(handler cluster.MembershipStateChangeHand
 	return id
 }
 
+// AddNearCacheConfig adds a near cache configuration.
+func (c *Config) AddNearCacheConfig(cfg nearcache.Config) {
+	c.ensureNearCacheConfigs()
+	c.NearcacheConfigs[cfg.Name] = cfg
```

Comment:
> Is there a reason that we use that additional nearCacheNames instead of iterating the keys of NearcacheConfigs?

I think to cache the names and not to have to reiterate the nearcacheConfigs map any time `GetNearCacheConfig` is called

Reply:
Serkan's reasoning is correct.

Reply:
I think it is okay to iterate it instead of having this extra field. Java client also does it. Also in that case I would argue "Validate" name does not describe the function

Reply:
I've removed `nearCacheNames` at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/cb30b44ad2e9338a11a4aa87138fc3dfb2fd017f

Reply:
addressed in [cb30b44](https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/cb30b44ad2e9338a11a4aa87138fc3dfb2fd017f) resolving